vignettes/grouping.Rmd

```diff
@@ -0,0 +1,172 @@
+---
+title: "Grouping behavior in yardstick"
```

Comment:
I had originally thought maybe this vignette would be called "Group-wise metrics" but it ended up being a bit more general than that.

Reply:
with a title like that, we might wanna preface a little earlier that this isn't about grouped data sets

Reply:
with a title like the one currently suggested or the previous "Group-wise metrics"? and in some ways, this is indeed about grouped data.

Reply:
i get worried, and confused myself a little bit, when we have this new methods, that share close names with group_by() related stuff. 

I was thinking of having a "when we say X we talk about ABC, and when we say Y we talk about XYZ"